Explorer/Assets/DCL/SDKComponents/AvatarShape/Systems/AvatarShapeHandlerSystem.cs

```diff
@@ -89,7 +100,108 @@ private void UpdateAvatarShape(ref PBAvatarShape pbAvatarShape, ref SDKAvatarSha
+        {
+            bool isSceneEmote = emoteId.EndsWith(".glb", StringComparison.OrdinalIgnoreCase);
+            if (!isSceneEmote) // normal "urn emote" or "base emote"
+            {
+                globalWorld.AddOrSet(globalWorldEntity,
+                    new CharacterEmoteIntent { EmoteId = emoteId });
+
+                return;
+            }
+
+            if (!sceneData.SceneContent.TryGetHash(emoteId, out string hash))
+            {
+                ReportHub.LogError(ReportCategory.AVATAR,$"Scene emote '{emoteId}' not found in scene assets. Aborting scene emote playing on SDK AvatarShape");
+                return;
+            }
+
+            if (globalWorld.TryGet(globalWorldEntity, out SDKAvatarShapeEmotePromiseCancellationToken? promiseComponent))
+                promiseComponent!.Dispose();
+
+            var newPromiseCancellationTokenComponent = new SDKAvatarShapeEmotePromiseCancellationToken();
+            globalWorld.AddOrSet(globalWorldEntity, newPromiseCancellationTokenComponent);
+
+            if (localSceneDevelopment)
+            {
+                LoadAndTriggerSceneEmoteAsync(globalWorldEntity,
+                    new GetSceneEmoteFromLocalSceneIntention(
+                        sceneData,
+                        emoteId,
+                        hash,
+                        bodyShape,
+                        loop: false), // looping scene emotes on SDK AvatarShapes is not supported yet
+                    newPromiseCancellationTokenComponent
+                ).Forget();
+            }
+            else
+            {
+                LoadAndTriggerSceneEmoteAsync(globalWorldEntity,
+                    new GetSceneEmoteFromRealmIntention(
+                        sceneData.SceneEntityDefinition.id!,
+                        sceneData.AssetBundleManifest,
+                        hash,
+                        loop: false, // looping scene emotes on SDK AvatarShapes is not supported yet
+                        bodyShape),
+                    newPromiseCancellationTokenComponent
+                ).Forget();
+            }
+        }
+
+        private async UniTaskVoid LoadAndTriggerSceneEmoteAsync<TIntention>(Entity globalWorldEntity, TIntention intention, SDKAvatarShapeEmotePromiseCancellationToken promiseComponent)
+            where TIntention : struct, IAssetIntention, IEquatable<TIntention>
+        {
+            CancellationToken ct = promiseComponent.Cts.Token;
+            try
+            {
+                // 1. Create the scene emote loading promise and wait for it
+                var promise = AssetPromise<EmotesResolution, TIntention>.Create(globalWorld, intention, PartitionComponent.TOP_PRIORITY);
+
+                Result<AssetPromise<EmotesResolution, TIntention>> promiseResult = await promise.ToUniTaskAsync(globalWorld, cancellationToken: ct).SuppressToResultAsync(ReportCategory.AVATAR);
```

Comment:
Avoid async flows in ECS, it's very error-prone - just introduce a separate query just like it's done for everything else